plugins/catalog-backend/src/modules/core/BuiltinKindsEntityProcessor.ts

```diff
@@ -125,6 +144,7 @@ export class BuiltinKindsEntityProcessor implements CatalogProcessor {
             },
             type: incomingRelation,
             target: selfRef,
+            metadata: target.metadata,
```

Comment:
Should metadata be bidirectional? Does metadata on the incoming relation mean the same thing as the outgoing? I don't _think_ so. `component.deploysToEnvironment` overrides wouldn't be quite the same as `environment.environmentHasDeployment`.